torchtune/models/llama3_2_vision/_component_builders.py

```diff
@@ -432,11 +447,22 @@ def lora_llama3_2_vision_encoder(
         "num_hidden_inputs": len(clip_hidden_states or []),
     }
     if fusion_lora:
-        projection_head = lora_llama3_2_vision_projection_head(**projection_options, **lora_options)
+        projection_head = lora_llama3_2_vision_projection_head(
+            **projection_options, **lora_options
+        )
     else:
         projection_head = lora_llama3_2_vision_projection_head(**projection_options)
 
-    return Llama3VisionEncoder(clip=clip, projection_head=projection_head)
+    encoder = Llama3VisionEncoder(clip=clip, projection_head=projection_head)
+
+    if quantize_base:
+        # For QLoRA, we reparametrize 4-bit tensors to bf16, and offload to CPU on the fly
+        # so as to not increase peak memory
+        encoder._register_state_dict_hook(
+            partial(reparametrize_as_dtype_state_dict_post_hook, offload_to_cpu=True)
+        )
+
+    return encoder
```

Comment:
A bunch of miscellaneous linter changes in this file, this is the only one of substance